tfx/components/evaluator/executor.py

```diff
@@ -176,52 +180,60 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
           eval_saved_model_path=model_path,
           add_metrics_callbacks=add_metrics_callbacks))
 
-    file_pattern = io_utils.all_files_pattern(
-        artifact_utils.get_split_uri(input_dict[constants.EXAMPLES_KEY], 'eval')
-    )
     eval_shared_model = models[0] if len(models) == 1 else models
     schema = None
     if constants.SCHEMA_KEY in input_dict:
       schema = io_utils.SchemaReader().read(
           io_utils.get_only_uri_in_dir(
               artifact_utils.get_single_uri(input_dict[constants.SCHEMA_KEY])))
 
+    # Load and deserialize example splits from execution properties.
+    example_splits = json_utils.loads(
+        exec_properties.get(constants.EXAMPLE_SPLITS_KEY, 'null')) or []
+    if not isinstance(example_splits, list):
+      raise ValueError('example_splits in execution properties needs to be a '
+                       'list. Got %s instead.' % type(example_splits))
+
     absl.logging.info('Evaluating model.')
     with self._make_beam_pipeline() as pipeline:
+      examples_list = []
+      tensor_adapter_config = None
       # pylint: disable=expression-not-assigned
-      if _USE_TFXIO:
-        tensor_adapter_config = None
-        if tfma.is_batched_input(eval_shared_model, eval_config):
+      if _USE_TFXIO and tfma.is_batched_input(eval_shared_model, eval_config):
+        for split in example_splits:
+          file_pattern = io_utils.all_files_pattern(
+              artifact_utils.get_split_uri(input_dict[constants.EXAMPLES_KEY],
+                                           split))
           tfxio = tf_example_record.TFExampleRecord(
               file_pattern=file_pattern,
               schema=schema,
               raw_record_column_name=tfma_constants.ARROW_INPUT_COLUMN)
-          if schema is not None:
-            tensor_adapter_config = tensor_adapter.TensorAdapterConfig(
-                arrow_schema=tfxio.ArrowSchema(),
-                tensor_representations=tfxio.TensorRepresentations())
-          data = pipeline | 'ReadFromTFRecordToArrow' >> tfxio.BeamSource()
-        else:
-          data = pipeline | 'ReadFromTFRecord' >> beam.io.ReadFromTFRecord(
-              file_pattern=file_pattern)
-        (data
-         | 'ExtractEvaluateAndWriteResults' >>
-         tfma.ExtractEvaluateAndWriteResults(
-             eval_shared_model=models[0] if len(models) == 1 else models,
-             eval_config=eval_config,
-             output_path=output_uri,
-             slice_spec=slice_spec,
-             tensor_adapter_config=tensor_adapter_config))
+          data = (pipeline
+                  | 'ReadFromTFRecordToArrow[%s]' % split >>
+                  tfxio.BeamSource())
+          examples_list.append(data)
+        if schema is not None:
+          tensor_adapter_config = tensor_adapter.TensorAdapterConfig(
+              arrow_schema=tfxio.ArrowSchema(),
```

Comment:
just noticed this depends on the field in the for loop, then it might not be correct, we probably need to enable tf_example_record.TFExampleRecord to take a list of file_pattern first, let's check with Zhuo about this

Reply:
yes, i recognized arrow_schema depends on tfxio and here it's always using the tfxio with the last file pattern in the list. 

if tf_example_record.TFExampleRecord can take in a list of file_pattern, then this tfxio will be correct to generate the arrow_schema.